cmake/modules/SwiftSource.cmake

```diff
@@ -383,29 +390,66 @@ function(_compile_swift_files dependency_target_out_var_name)
     #
     # To work around this issue we touch the output files so that their mtime
     # always gets updated.
-    set(command_touch_outputs
-      COMMAND "${CMAKE_COMMAND}" -E touch ${outputs})
+    set(command_touch_standard_outputs
+      COMMAND "${CMAKE_COMMAND}" -E touch ${standard_outputs})
+    set(command_touch_apinotes_outputs
+      COMMAND "${CMAKE_COMMAND}" -E touch ${apinotes_outputs})
+    set(command_touch_module_outputs
+      COMMAND "${CMAKE_COMMAND}" -E touch ${module_outputs})
+  endif()
+
+  # First generate the api notes if we need them.
+  list(LENGTH command_create_apinotes num_apinotes_outputs)
```

Comment:
if (apinotes_outputs) is much better.  I'll use that.  I just don't know enough CMake to know if thats ok or not.  I figured it would be on a string but wasn't sure about a list.